src/status_im2/common/toasts/view.cljs

```diff
@@ -1,89 +1,56 @@
-
-(defn dismiss
-  [translate-y dismissed-locally? close!]
-  (reanimated/animate-shared-value-with-spring
-   translate-y
-   -500
-   {:mass 1 :damping 20 :stiffness 300})
-  (reset! dismissed-locally? true)
-  (close!))
-
 (defn f-container
-  [id]
-  (let [dismissed-locally? (reagent/atom false)
-        close!             #(rf/dispatch [:toasts/close id])
-        timer              (reagent/atom nil)
-        clear-timer        #(background-timer/clear-timeout @timer)]
+  [toast-id]
+  (let [dismissed-locally?    (reagent/atom false)
+        set-dismissed-locally #(reset! dismissed-locally? true)
+        close-toast           #(rf/dispatch [:toasts/close toast-id])
+        timer                 (reagent/atom nil)
+        clear-timer           #(background-timer/clear-timeout @timer)]
     (fn []
-      (let [duration (or (rf/sub [:toasts/toast-cursor id :duration]) 3000)
-            on-dismissed (or (rf/sub [:toasts/toast-cursor id :on-dismissed]) identity)
-            create-timer (fn []
-                           (reset! timer (background-timer/set-timeout close! duration)))
-            translate-y (reanimated/use-shared-value 0)
-            pan
-            (->
-              (gesture/gesture-pan)
-              (gesture/on-start clear-timer)
-              (gesture/on-update
-               (fn [^js evt]
-                 (let [evt-translation-y (.-translationY evt)
-                       pan-down?         (> evt-translation-y 100)
-                       pan-up?           (< evt-translation-y -30)]
-                   (cond
-                     pan-down? (reset-translate-y translate-y)
-                     pan-up? (dismiss translate-y dismissed-locally? close!)
-                     :else
-                     (reanimated/set-shared-value translate-y
-                                                  evt-translation-y)))))
-              (gesture/on-end (fn [_]
-                                (when-not @dismissed-locally?
-                                  (reanimated/set-shared-value translate-y 0)
-                                  (create-timer)))))]
+      (let [duration     (or (rf/sub [:toasts/toast-cursor toast-id :duration]) 3000)
+            on-dismissed (or (rf/sub [:toasts/toast-cursor toast-id :on-dismissed]) identity)
+            create-timer #(reset! timer (background-timer/set-timeout close-toast duration))
+            translate-y  (reanimated/use-shared-value 0)
+            pan-gesture  (animation/pan-gesture {:clear-timer           clear-timer
+                                                 :create-timer          create-timer
+                                                 :translate-y           translate-y
+                                                 :close-toast           close-toast
+                                                 :set-dismissed-locally set-dismissed-locally
+                                                 :dismissed-locally?    @dismissed-locally?})]
```

Comment:
Mainly moved this logic to animation namespace to make it easier to read and separate animation stuff